microsoft/atlas-design
# add single and holy-grail layout component, update site

Tasks:
- task-925512
- task-925516

Link: preview-582
Layout page link: https://design.learn.microsoft.com/pulls/582/components/layout.html

This pull request adds a new layout component to Atlas and that components first two layouts (of 4 initially planned).

Layouts in this PR are single and holy grail. The simplest and the most complex of those planned.

In order to better test the CSS changes (and to take advantage of the new layout component itself), I retooled the site slightly.

- applied the holy grail layout and removed some of the one-off site code
- a mini feature for applying different layouts on the layout documentation page.
- a mobile menu was added to ensure that on narrow widths we could hide the large menu of navigation links. This feature was done quite hastily, and its approach was that way it is because we've yet to create a modal in Atlas. I would like to revisit it when modal is added to further improve it and remove the fullscreen functionality.
- as part of the above, a burger component that lives in the site only. Perhaps it can be removed as nav bar moves into Atlas.
- In order to test the layout-hero container, I added some build logic to extract the H1 and first paragraph from the markdown article and then add them to a hero element. This is done with yaml front matter, and `hero: true`. I only applied it to the layout page for now.

## Testing

1. Visit preview link above. Visit `/components/layout.html` in TOC.
2. Resize screen to narrow. Open new mobile menu.
3. On the layout component page, try buttons in the "Try" section near the top. Also apply single layout.
4. Resize screen.
6. See paired PR.

## Contributor checklist
